lib/src/cpp/common/LocatorList.cpp

```diff
@@ -162,12 +164,85 @@ void set_kind(
 
 void set_port(
         utils::ParseXML& manager,
-        xercesc::DOMElement& xml_node,
+        xercesc::DOMNode* xml_node,
         const std::string& port,
         const std::string& index,
         const bool is_external)
 {
-    throw Unsupported("Unsupported");
+    xercesc::DOMDocument* doc = manager.get_doc();
+    xercesc::DOMNode* locator_node = nullptr;
+    xercesc::DOMNode* kind_node = nullptr;
+    xercesc::DOMNode* port_node = nullptr;
+
+    // Create new locator and push it in the list
+    if (index.empty())
+    {
+        // create default udp v4 kind
+        kind_node = static_cast<xercesc::DOMNode*> (doc->createElement(
+                    xercesc::XMLString::transcode(utils::tag::UDP_V4_LOCATOR)));
+
+        // append the new kind node directly to the given parent
+        if (is_external)
+        {
+            xml_node->appendChild(kind_node);
+        }
+        // create <locator> tag if required
+        else
+        {
+            locator_node = static_cast<xercesc::DOMNode*> (doc->createElement(
+                        xercesc::XMLString::transcode(utils::tag::LOCATOR)));
+            xml_node->appendChild(locator_node);
+            locator_node->appendChild(kind_node);
+        }
+
+        // add port node to the kind node
+        port_node = static_cast<xercesc::DOMNode*> (doc->createElement(
+                    xercesc::XMLString::transcode(utils::tag::PORT)));
+        kind_node->appendChild(port_node);
+    }
+    // set the port value in the locator defined by the index
+    else
+    {
+        std::string* ind = new std::string();
```

Comment:
With the infrastructure refactor, I suggest avoiding the pointer and directly allocate the string object, unless there is a specific reason to have a pointer initialized.